core/runtime.rs

```diff
@@ -351,6 +360,42 @@ impl JsRuntime {
       .unwrap();
   }
 
+  /// Initializes JS of provided Extensions
+  // NOTE: this will probably change when streamlining snapshot flow
+  pub fn init_extension_js(&mut self) -> Result<(), AnyError> {
+    // Take extensions to avoid double-borrow
+    let mut extensions: Vec<Extension> = self.extensions.drain(..).collect();
+    for m in extensions.iter_mut() {
+      let js_files = m.init_js();
+      for (filename, source) in js_files {
+        // TODO(@AaronO): use JsRuntime::execute_static() here to move src off heap
+        self.execute(filename, source)?;
+      }
+    }
+    self.extensions = extensions;
+    Ok(())
+  }
+
+  /// Initializes ops of provided Extensions
+  // NOTE: this will probably change when streamlining snapshot flow
+  pub fn init_extension_ops(&mut self) -> Result<(), AnyError> {
```

Comment:
I'm thinking that maybe this method should consume the `extensions` field? It shouldn't really be used after initializing the ops, right?

Reply:
Plugins might be implemented as "dynamic extensions", in which case tracking them post init will streamline cleaning them up if we want to allow unloading plugins (which would make sense)